### PR TITLE
create DBFlute demo project (all requests and tests can work)

### PR DESCRIPTION
**Pull request by jflute**
o auto-generator (DBFlute Client and Engine) is git-ignored
o make .gitignore for demo-dbflute project (ignoring Generator resources and Eclipse settings)

### CODE REVIEW

File: demo-dbflute/src/main/java/com/example/PizzaRepository.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        });
+        pizzaBhv.loadPizzaToppings(pizzaList, toppingsCB -> {
+            toppingsCB.setupSelect_Topping();
+        });
+        return pizzaList.mappingList(pizza -> { // mapping entity to model
+            com.example.model.Pizza modelPizza = new com.example.model.Pizza();
+            modelPizza.setId(pizza.getId());
+            modelPizza.setName(pizza.getName());
+            modelPizza.setPrice(pizza.getPrice());
+            pizza.getBase().ifPresent(base -> { // because BASE_ID is null-able
```
Comment (making) [inline demo-dbflute/src/main/java/com/example/PizzaRepository.java]: Sorry for my lazy schema. `BASE_ID` is actually non-null.

File: demo-dbflute/src/main/java/com/example/DemoDBFluteApplication.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.example.dbflute.allcommon.DBFluteBeansJavaConfig;
+import com.example.model.Base;
+import com.example.model.Pizza;
+import com.example.model.Topping;
+
+import bitronix.tm.BitronixTransactionSynchronizationRegistry;
+
+@SpringBootApplication
+@Import(DBFluteBeansJavaConfig.class) // #dbflute
```
Comment (making) [inline demo-dbflute/src/main/java/com/example/DemoDBFluteApplication.java]: I believe `@Import` is not necessary. `DBFluteBeansJavaConfig` can be scanned.

**Comment by jflute** [comment]
I see. I added not-null constraint to BASE_ID and regenerate it.

File: demo-dbflute/src/main/java/com/example/DemoDBFluteApplication.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            pizza.setBase(new Base(1L));
+            pizza.setToppings(Arrays.asList(new Topping(1L), new Topping(2L), new Topping(3L), new Topping(4L), new Topping(5L)));
+            pizza.setName("Hello Pizza");
+            pizza.setPrice(new BigDecimal("1000"));
+            pizzaRepository.save(pizza);
+        };
+    }
+
+    @Bean
+    TransactionSynchronizationRegistry synchronizationRegistry(TransactionManager transactionManager) {
```
Comment (making) [inline demo-dbflute/src/main/java/com/example/DemoDBFluteApplication.java]: Is this necessary?

**Comment by jflute** [comment]
> I believe @Import is not necessary. DBFluteBeansJavaConfig can be scanned.

Oh, OK. Certainly It can work without import. I removed it.

**Comment by jflute** [comment]
> Is this necessary?

Sorry, I merely copy runner() from Mybatis project,
and synchronizationRegistry() from S2JDBC project.
(because DBFlute is similar to S2JDBC about transaction)

But certainly it can work without the synchronizationRegistry().
Now, I've found the jdbcManager uses it in S2JDBC project.
Anyway, I removed synchronizationRegistry().

**Comment by making** [comment]
Awesome! Thanks!

**Pull request merged**